ReplayKitExample/ReplayKitExample/ReplayKitVideoSource.swift

```diff
@@ -65,29 +99,54 @@ class ReplayKitVideoSource: NSObject, VideoSource {
         }
     }
 
-    static func formatRequestToDownscale(maxWidthOrHeight: Int) -> VideoFormat {
+    static private func formatRequestToDownscale(maxWidthOrHeight: UInt, maxFrameRate: UInt) -> VideoFormat {
         let outputFormat = VideoFormat()
 
         var screenSize = UIScreen.main.bounds.size
         screenSize.width *= UIScreen.main.nativeScale
         screenSize.height *= UIScreen.main.nativeScale
 
-        let downscaledTarget = CGSize(width: maxWidthOrHeight,
-                                      height: maxWidthOrHeight)
-        let fitRect = AVMakeRect(aspectRatio: screenSize,
-                                 insideRect: CGRect(origin: CGPoint.zero, size: downscaledTarget)).integral
-        let outputSize = fitRect.size
+        if maxWidthOrHeight > 0 {
+            let downscaledTarget = CGSize(width: Int(maxWidthOrHeight),
+                                          height: Int(maxWidthOrHeight))
+            let fitRect = AVMakeRect(aspectRatio: screenSize,
+                                     insideRect: CGRect(origin: CGPoint.zero, size: downscaledTarget)).integral
+            let outputSize = fitRect.size
+
+            outputFormat.dimensions = CMVideoDimensions(width: Int32(outputSize.width), height: Int32(outputSize.height))
+        } else {
+            outputFormat.dimensions = CMVideoDimensions(width: Int32(screenSize.width), height: Int32(screenSize.height))
+        }
 
-        outputFormat.dimensions = CMVideoDimensions(width: Int32(outputSize.width), height: Int32(outputSize.height))
+        outputFormat.frameRate = maxFrameRate
         return outputFormat;
     }
 
+    static func getParametersForUseCase(codec: VideoCodec, isScreencast: Bool) -> (EncodingParameters, VideoFormat) {
```

Comment:
It would be nice to have the device model as another input, and then cap at 15 fps on older devices. Don't bother with the IVTC & trying to send 24 fps content.